src/login/logind-seat.c

```diff
@@ -225,8 +225,21 @@ int seat_set_active(Seat *s, Session *session) {
         assert(s);
         assert(!session || session->seat == s);
 
-        if (session == s->active)
+        /* When logind receives the SIGRTMIN signal from the kernel, it will
+         * execute session_leave_vt and stop all devices of the session; at
+         * this time, if the session is active and there is no change in the
+         * session, then the session does not have the permissions of the device,
+         * and the machine will have a black screen and suspended animation.
+         * Therefore, if the active session has executed session_leave_vt ,
+         * A resume is required here. */
+        if (session == s->active) {
+                if (session && session->left) {
+                        log_debug("Active but the session remains unchanged,resume the session that has been left.");
+                        session_device_resume_all(session);
+                        session->left = false;
```

Comment:
hmm, why maintain the boolean? why not just call session_device_resume_all() unconditionally?

i mean, the call is a NOP anyway for devices already resumed. Hence it's always safe to just call this, and it will unpause everything that might be resume or be a NOP.

Reply:
Thank you, I have made modifications according to your comments,please review